src/scene/materials/standard-material.js

```diff
@@ -97,13 +111,26 @@ Object.assign(pc, function () {
      * @property {boolean} opacityVertexColor Use mesh vertex colors for opacity. If opacityMap is set, it'll be multiplied by vertex colors.
      * @property {string} opacityVertexColorChannel Vertex color channels to use for opacity. Can be "r", "g", "b" or "a".
      *
-     * @property {pc.Texture|null} normalMap The normal map of the material (default is null).
+     * @property {pc.Texture|null} normalMap The main (primary) normal map of the material (default is null).
      * The texture must contains normalized, tangent space normals.
-     * @property {number} normalMapUv Normal map UV channel.
-     * @property {pc.Vec2} normalMapTiling Controls the 2D tiling of the normal map.
-     * @property {pc.Vec2} normalMapOffset Controls the 2D offset of the normal map. Each component is between 0 and 1.
-     * @property {number} bumpiness The bumpiness of the material. This value scales the assigned normal map.
+     * @property {number} normalMapUv Main (primary) normal map UV channel.
+     * @property {pc.Vec2} normalMapTiling Controls the 2D tiling of the main (primary) normal map.
+     * @property {pc.Vec2} normalMapOffset Controls the 2D offset of the main (primary) normal map. Each component is between 0 and 1.
+     * @property {number} bumpiness The bumpiness of the material. This value scales the assigned main (primary) normal map.
+     * It should be normally between 0 (no bump mapping) and 1 (full bump mapping), but can be set to e.g. 2 to give even more pronounced bump effect.
+     *
+     * @property {pc.Texture|null} normalDetailMap The detail (secondary) normal map of the material (default is null). Will only be used if main (primary) normal map is non-null.
+     * @property {number} normalDetailMapUv Detail (secondary) normal map UV channel.
+     * @property {pc.Vec2} normalDetailMapTiling Controls the 2D tiling of the detail (secondary) normal map.
+     * @property {pc.Vec2} normalDetailMapOffset Controls the 2D offset of the detail (secondary) normal map. Each component is between 0 and 1.
+     * @property {number} normalDetailMapBumpiness The bumpiness of the material. This value scales the assigned detail (secondary) normal map.
      * It should be normally between 0 (no bump mapping) and 1 (full bump mapping), but can be set to e.g. 2 to give even more pronounced bump effect.
+     * @property {string} normalDetailBlend Determines how the detail (secondary) and detail (secondary) diffuse maps are blended together. Can be:
+     * * {@link pc.DETAILBLEND_NORMALS_RNM}: Blend normals using Reoriented Normal Mapping. Slowest but also most accurate.
+     * * {@link pc.DETAILBLEND_NORMALS_PD}: Blend normals using Partial Derivatives. Cheaper but less accurate than {@link pc.DETAILBLEND_NORMALS_RNM}.
+     * * {@link pc.DETAILBLEND_NORMALS_WHITEOUT}: Blend normals based on AMD's ruby Whiteout demo. Slightly cheaper than {@link pc.DETAILBLEND_NORMALS_PD}.
```

Comment:
I'm not keen on the naming here. Naming API enums after demos or companies isn't great. Is there not a descriptive alternative that is specific to the actual math/algorithm?

Reply:
I'm not fond of them either. Unfortunately,  as far as I can tell, they are only names our industry knows them by. I'm happy to change them, but it will only cause confusion for those in the know.

**Sources:**
- [MATERIAL ADVANCES IN CALL OF DUTY: WWII:
![image](https://user-images.githubusercontent.com/255073/79243350-cd5e3c80-7e75-11ea-839e-1182f6a5c6c2.png)](https://www.activision.com/cdn/research/MaterialAdvancesInWWII.pdf)
- [Normal Mapping for a Triplanar Shader:
![image](https://user-images.githubusercontent.com/255073/79243414-e109a300-7e75-11ea-857b-8b519eb68de5.png)](https://medium.com/@bgolus/normal-mapping-for-a-triplanar-shader-10bf39dca05a#a969)
- [Mixing Normal Maps:
![image](https://user-images.githubusercontent.com/255073/79245330-41014900-7e78-11ea-9427-8a54c77fd01b.png)](http://oliverm-h.blogspot.com/2014/01/ue3udk-mixing-normal-maps.html)
- [Blending in Detail:
![image](https://user-images.githubusercontent.com/255073/79242967-4c06aa00-7e75-11ea-945a-9a537b165030.png)](https://blog.selfshadow.com/publications/blending-in-detail/)

Reply:
That does help. But head [here](https://udn.unrealengine.com/users/login.html?orig=true) and read the footer. Here's my question to you: are you genuinely going to notice the performance different between these 4 options? In the screenshot above, the difference in the number of shader instructions is small. Why not just ship RNM (since it's the best looking) and keep the API simple?

Reply:
The `here` link only points to `https://udn.unrealengine.com/users/login.html?orig=true`

I do like the idea of keeping the API simple. I would suggest keeping the most accurate (RNM) and the least expensive (UDN). Depending on an artist's previous workflow they might also be more familiar (or already have existing assets) with a common method in mind - which would likely be either UDN/Whiteout (oldest and most visually similar) or RNM. Performance is still the stronger argument, but maybe the link you mention suggests otherwise.

Reply:
The footer says "Epic, Epic Games, Unreal, Unreal Developer Network, *UDN*, Unreal Engine and UE4 *are trademarks or registered trademarks of Epic Games, Inc.* in the United States of America and elsewhere."

I would suggest just going with RNM for now. The use case for detail maps is probably going to be 3D visualizers/configurators for the most part, at least to begin with - so quality will matter above fps. But frankly, I strongly believe a small number of fragment shader instructions difference is going to make a tiny difference. If you can prove it *does* matter with a subsequent PR and provide some numbers to demonstrate, that'd be great.

Reply:
Alright, will fix later in the week.

Reply:
Thanks @aidinabedi - and I promise to merge as soon as that change lands. 😄  Sorry to always be such a pain.

Reply:
No worries :)

Reply:
Fixed

Reply:
Perfect, thanks.